⎕NA: Add macOS to System Shared Libraries section (#920)
Dyalog/documentation: unix-installation-and-configuration-guide/docs/quadna.md

# `⎕NA` under UNIX

`⎕NA` is fully supported under all supported non-Windows platforms; the Conga communications package for example is a shared library on all platforms.

> `⎕NA` supports user-written shared libraries and also system-supplied shared libraries. Dyalog APL under Non-Windows platforms is supplied with a shared library, dyalog32 or dyalog64 which contains the same functions as the dyalog32.dll and dyalog64.dll as described in the [`⎕NA` documentation](../../language-reference-guide/system-functions/na#the-dyalog-dll). The file extension of the shared library is operating system dependent. Additionally, the function `getlasterror` is included; this returns the error code at the point when the called function failed (which may be different from its value at the point where a previous error occurred).

Reviewer: ```suggestion
`⎕NA` supports user-written shared libraries and system-supplied shared libraries. Under non-windows platforms, Dyalog is supplied with a shared library, **dyalog32** or **dyalog64**, that contains the same functions as **dyalog32.dll** and **dyalog64.dll** (as described in [`⎕NA`: The Dyalog DLL](../../language-reference-guide/system-functions/na#the-dyalog-dll)); the file extension of the shared library is operating-system dependent. The function `getlasterror` is also included – this returns the error code at the point when the called function failed, which can be different from its value at the point when a previous error occurred).
```

> `⎕NA` is fully supported under UNIX; the Conga communications package for example is a shared library on all platforms.

Reviewer 2: ```suggestion
`⎕NA` is fully supported under all supported non-Windows platforms; the Conga communications package for example is a shared library on all platforms.
```

> `⎕NA` supports user-written shared libraries and also system-supplied shared libraries. Dyalog APL under Non-Windows platforms is supplied with a shared library, dyalog32.so or dyalog64.so which contains the same functions as the DLLs which are described in the `⎕NA` documentation in the *Dyalog Language Reference Guide*. Additionally, the function `getlasterror` is included; this returns the error code at the point when the called function failed (which may be different from its value at the point where a previous error occurred).

Reviewer 2: ```suggestion
`⎕NA` supports user-written shared libraries and also system-supplied shared libraries. Dyalog APL under Non-Windows platforms is supplied with a shared library, dyalog32 or dyalog64 which contains the same functions as the dyalog32.dll and dyalog64.dll which are described in the "The Dyalog DLL" section of the `⎕NA` documentation in the *Dyalog Language Reference Guide*.  The file extension of the shared library is operating system dependent. Additionally, the function `getlasterror` is included; this returns the error code at the point when the called function failed (which may be different from its value at the point where a previous error occurred).
```

### Definitions

> In the remainder of this section references are made to the APL variables `sharedlib` and `dyalib`; the definitions for both vary between operating system, and between 32 and 64 bit interpreters.

Reviewer: ```suggestion
In the remainder of this section, references are made to the APL variables `sharedlib` and `dyalib`; the definitions for these vary between operating system and between 32- and 64-bit interpreters.
```

      sharedlib←libc ⍬
```

Under macOS, `sharedlib` is defined as:
```apl
      sharedlib←'libc.dylib'
```

For AIX and all Linux platforms, the dyalog shared library is identified as
```apl
      dyalib←'dyalog64.so'         ⍝ 64 bit
      dyalib←'dyalog32.so'         ⍝ 32 bit
	
```

For macOS, the dyalog shared library is identified as 
```apl
      dyalib←dyalog64.dylib
```

> The Setup function in the NonWindows namespace in the quadna workspace can be used to set both sharedlib and dyalib for all supported non-Windows platforms.

Reviewer: ```suggestion
The `Setup` function in the `NonWindows` namespace in the `quadna` workspace can be used to set both `sharedlib` and `dyalib` for all supported non-Windows platforms.
```

## geterrno

> The dyalog shared libary under Non-Windows platforms includes the function `geterrno`. This returns the current value of errno; be aware that it may not have the same value as at the point when the error was raised. To use this function:

Reviewer 2: ```suggestion
The dyalog shared library under Non-Windows platforms includes the function `geterrno`. This returns the current value of errno; be aware that it may not have the same value as at the point when the error was raised. To use this function:
```
